src/gen-s-parser.inc

```diff
@@ -0,0 +1,688 @@
+// DO NOT EDIT! This file generated by scripts/gen-s-parser.py
+
+#ifdef INSTRUCTION_PARSER
+#undef INSTRUCTION_PARSER
```

Comment:
Why both with this guard?

Reply:
This is the same way includes of TableGen-generated code works. `#undef`ing the macro here means that clients don't need to worry about doing it after `#include`ing the generated results. Strictly speaking, this guard is completely unnecessary since right now you either include the entire file or none of it. I could remove it if it seems distracting.